tasks/collect-data/collect-data.yaml

```diff
@@ -76,45 +76,46 @@ spec:
           value: '$(params.snapshot)'
       script: |
         #!/usr/bin/env sh
+        # shellcheck disable=SC3040
         set -eo pipefail
 
         RESULTS_DIR_PATH="results"
         if [ -n "$(params.subdirectory)" ]; then
-          mkdir -p $(workspaces.data.path)/$(params.subdirectory)
+          mkdir -p "$(workspaces.data.path)/$(params.subdirectory)"
           RESULTS_DIR_PATH="$(params.subdirectory)/results"
         fi
 
         mkdir "$(workspaces.data.path)/$RESULTS_DIR_PATH"
-        echo -n $RESULTS_DIR_PATH > $(results.resultsDir.path)
+        printf "%s" "$RESULTS_DIR_PATH" > "$(results.resultsDir.path)"
```

Comment:
Why these changes from echo to printf? If it's this:
```
^-- [SC3037](https://www.shellcheck.net/wiki/SC3037) (warning): In POSIX sh, echo flags are undefined.
```
Then again, using `bash` instead of `sh` should help.

Reply:
Thanks.